File: morphStream/morph-web/dashboard/src/app/model/Operator.ts
export interface Operator {
  id: number;
  name: string;
  numOfInstances: number;
  throughput: number; // tuples/s
  latency: number;  // ms
  explorationStrategy: string;
  schedulingGranularity: string;
  abortHandling: string;
  numOfTD: number;
  numOfLD: number;
  numOfPD: number;
  lastBatch: number;
}

File: morph-common/src/main/java/dao/Operator.java
package dao;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;

@Data
@ToString(callSuper = true)
@JsonIgnoreProperties(ignoreUnknown = true)
@AllArgsConstructor
@NoArgsConstructor
public class Operator {
    private String id;
    private String name;
    private int numOfInstances;
    private double throughput;
    private double latency;
    private String explorationStrategy;
    private String schedulingGranularity;
    private String abortHandling;

    public Operator(String id, String name, int numOfInstances) {
        this.id = id;
        this.name = name;
        this.numOfInstances = numOfInstances;
    }
}
